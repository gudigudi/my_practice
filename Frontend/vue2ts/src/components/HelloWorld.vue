<template>
  <div class="hello">
    <h1>
      <span>{{ msg }}</span>
    </h1>
    <p>count:{{count}}</p>
    <button @click="setCount(count+1)">+1</button>
    <button @click="setCount(count-1)">-1</button>
  </div>
</template>

<script lang="ts">
import { Component, Prop, PropSync, Vue } from "vue-property-decorator";

const store = Vue.observable({ count: 0 });

@Component
export default class HelloWorld extends Vue {
  @Prop(String) public readonly msg: string | undefined;

  get count() {
    return store.count;
  }

  public setCount(count: number) {
    store.count = count;
  }
}
</script>

<style lang="scss" scoped>
.hello {
  text-align: center;
  padding: 20px;
  font-size: 32px;
  color: #666;
  line-height: 56px;
  h1 {
    color: #333;
    line-height: 1.15;
    font-size: 64px;
    margin-bottom: 32px;
    span {
      position: relative;
      &::after {
        content: "";
        position: absolute;
        z-index: -1;
        left: 0;
        bottom: 8px;
        width: 100%;
        height: 20px;
        background: #ecf6ff;
      }
    }
  }
}
</style>
