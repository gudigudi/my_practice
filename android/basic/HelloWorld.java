public class HelloWorld extends android.app.Activity {

    public void create(android.os.Bundle.savedInstantceState) {
        super.onCreate(savedInstantceState);
        System.out.println("Hello, World!");
    }
}
