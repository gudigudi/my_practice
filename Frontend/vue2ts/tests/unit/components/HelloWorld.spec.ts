import { shallowMount, Wrapper } from '@vue/test-utils'
import HelloWorld from '@/components/HelloWorld.vue'

let wrapper: Wrapper<Vue>

afterEach(() => {
    //TODO reset prop "count"
    (wrapper.vm as any).setCount(0)
    wrapper.destroy();
});

describe('HelloWorld.vue', () => {
    it('renders props.msg when passed', () => {
        const msg = 'new message'
        wrapper = shallowMount(HelloWorld, {
            propsData: { msg }
        })
        expect(wrapper.props('msg')).toBe(msg)
        expect(wrapper.find('p').text()).toBe('count:' + '0')
        expect(wrapper).toMatchSnapshot()
    })
    it('getter/setter of prop count', async () => {
        wrapper = shallowMount(HelloWorld);

        await (wrapper.vm as any).setCount(10)

        expect(wrapper.find('p').text()).toBe('count:' + '10')
    })
    it('click button to add and sub count', async () => {
        wrapper = shallowMount(HelloWorld);

        const btns = wrapper.findAll('button')
        let btnAdd = btns.at(0)
        let btnSub = btns.at(1)

        await btnAdd.trigger('click')
        expect(wrapper.find('p').text()).toBe('count:' + '1')
        await btnSub.trigger('click')
        expect(wrapper.find('p').text()).toBe('count:' + '0')
    })
})
