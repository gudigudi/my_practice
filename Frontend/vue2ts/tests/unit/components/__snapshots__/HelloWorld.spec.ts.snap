// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HelloWorld.vue renders props.msg when passed 1`] = `
<div class="hello">
  <h1><span>new message</span></h1>
  <p>count:0</p> <button>+1</button> <button>-1</button>
</div>
`;
